pkg/cio/io.go

```diff
@@ -286,18 +296,27 @@ func BinaryIO(binary string, args map[string]string) Creator {
 
 // TerminalBinaryIO forwards container STDOUT|STDERR directly to a logging binary
 // It also sets the terminal option to true
-func TerminalBinaryIO(binary string, args map[string]string) Creator {
+func TerminalBinaryIO(binary string, args map[string]string, setStderr ...bool) Creator {
```

Comment:
We should probably just have a model similar to NewFIFOSetInDir, where the unix version doesn't do anything fancy, and the windows version has a special case for the terminal bool we pass in.

Unix:
```go
// NewFIFOSetInDir returns a new FIFOSet with paths in a temporary directory under root
func NewFIFOSetInDir(root, id string, terminal bool) (*FIFOSet, error) {
	if root != "" {
		if err := os.MkdirAll(root, 0700); err != nil {
			return nil, err
		}
	}
	dir, err := os.MkdirTemp(root, "")
	if err != nil {
		return nil, err
	}
	closer := func() error {
		return os.RemoveAll(dir)
	}
	return NewFIFOSet(Config{
		Stdin:    filepath.Join(dir, id+"-stdin"),
		Stdout:   filepath.Join(dir, id+"-stdout"),
		Stderr:   filepath.Join(dir, id+"-stderr"),
		Terminal: terminal,
	}, closer), nil
}
```

Windows:
```go
// NewFIFOSetInDir returns a new set of fifos for the task
func NewFIFOSetInDir(_, id string, terminal bool) (*FIFOSet, error) {
	stderrPipe := ""
	if !terminal {
		stderrPipe = fmt.Sprintf(`%s\ctr-%s-stderr`, pipeRoot, id)
	}
	return NewFIFOSet(Config{
		Terminal: terminal,
		Stdin:    fmt.Sprintf(`%s\ctr-%s-stdin`, pipeRoot, id),
		Stdout:   fmt.Sprintf(`%s\ctr-%s-stdout`, pipeRoot, id),
		Stderr:   stderrPipe,
	}, nil), nil
}
```

Reply:
For backward compatibility, we want to avoid introducing a new required argument that could potentially disrupt other applications using the method.

Reply:
If the method always sets terminal to true, and on windows we require that stderr is not filled in, why do we need a new parameter at all? The Windows version of the method can assign stderr to empty string, and unix can get away with whatever it does today right?

Reply:
Refactored.